### ISSUE TITLE
Focused links can be covered by the table of contents button on mobile

### ISSUE THREAD

**Issue by tombye** [alphagov/tech-docs-gem#356]
## What should change

The table of contents button should be changed for smaller screens so it can't overlap links when they are focused.

## User need

If you rely on the keyboard, you need to see what's in focus at all times. If a link gets hidden, it also breaks [success criterion 2.4.11, Focus not obscured (minimum)](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum.html) and [success criterion 2.4.12, Focus not obscured (enhanced)](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-enhanced.html).

For example, see the link to the 'GOV.UK kubernetes manual' on [https://docs.publishing.service.gov.uk/](https://docs.publishing.service.gov.uk/manual/architecture-deep-dive.html) receiving focus while hidden:

![mobile_tech_docs_template_focus_obscured_fail](https://github.com/user-attachments/assets/7bb09015-4f0d-473c-bd63-773a3af193ab)

**Comment by tombye** [comment]
We added a fix for this in https://github.com/alphagov/tech-docs-gem/pull/360 so I'm marking this as closed.
